src/box/applier.cc

```diff
@@ -207,7 +207,9 @@ applier_thread_writer_f(va_list ap)
 		try {
 			applier->thread.has_acks_to_send = false;
 			struct xrow_header xrow;
-			xrow_encode_vclock(&xrow, &applier->thread.ack_vclock);
+			xrow_encode_heartbeat(&xrow,
```

Comment:
Should be `xrow_encode_heartbeat_xc`?

Reply:
Yes, indeed. Fixed together with the next comment:

<details><summary>diff</summary>

```diff
diff --git a/src/box/applier.cc b/src/box/applier.cc
index 5e0a821d7..11b251293 100644
--- a/src/box/applier.cc
+++ b/src/box/applier.cc
@@ -207,9 +207,10 @@ applier_thread_writer_f(va_list ap)
                try {
                        applier->thread.has_acks_to_send = false;
                        struct xrow_header xrow;
-                       xrow_encode_heartbeat(&xrow,
-                                             &applier->thread.ack_vclock,
-                                             applier->thread.last_vclock_sync);
+                       uint64_t vclock_sync = applier->thread.last_vclock_sync;
+                       struct vclock *ack_vclock = &applier->thread.ack_vclock;
+                       xrow_encode_heartbeat_xc(&xrow, ack_vclock,
+                                                vclock_sync);
                        xrow.tm = applier->thread.txn_last_tm;
                        coio_write_xrow(&applier->io, &xrow);
                        ERROR_INJECT(ERRINJ_APPLIER_SLOW_ACK, {
@@ -797,10 +798,7 @@ applier_parse_tx_row(struct applier_tx_row *tx_row)
                        diag_raise();
                }
        } else if (type == IPROTO_OK) {
-               if (xrow_decode_heartbeat(row, NULL,
-                                         &tx_row->req.vclock_sync) != 0) {
-                       diag_raise();
-               }
+               xrow_decode_heartbeat_xc(row, NULL, &tx_row->req.vclock_sync);
        } else {
                tnt_raise(ClientError, ER_UNKNOWN_REQUEST_TYPE, type);
        }
```

</details>